### PR TITLE
bradesco: timeout after 5s

### PR DESCRIPTION
**Pull request by marquinhoalm**
add request provider timeout

closes pagarme/superbowleto#207

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel confortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

File: src/providers/bradesco/index.ts [hunk 1]
```diff
@@ -12,6 +12,11 @@ const { endpoint } = prop('bradesco', getConfig())
 
 const makeLogger = makeFromLogger('bradesco/index')
 
+const defaultValue = {
```
Comment (grvcoelho) [inline src/providers/bradesco/index.ts]: Renomeia isso para `defaultResponseCode`

File: src/providers/bradesco/index.ts [hunk 2]
```diff
@@ -125,6 +126,10 @@ export const register = (boleto) => {
     .tap((response) => {
       logger.info({ status: 'succeeded', metadata: { status: response.status, data: response.data } })
     })
+    .catch(Promise.TimeoutError, (e) => {
```
Comment (grvcoelho) [inline src/providers/bradesco/index.ts]: Renomeia `e` para `err`, para seguir o padrão de outros lugares.

File: test/functional/boleto/register/pending.js [hunk 3]
```diff
@@ -38,4 +32,34 @@ test('registers a boleto (provider pending)', async (t) => {
...
+      .delay(6000)
+  )
+
+  const payload = mock
+
+  const { body } = await create({
+    body: payload
+  })
+
+  const boleto = await register({
```
Comment (grvcoelho) [inline test/functional/boleto/register/pending.js]: Não deveríamos usar `provider.register` aqui? Ao invés de register?

Comment (marquinhoalm) [inline test/functional/boleto/register/pending.js]: Era apenas com o register mesmo. 
Era o ordem do delay que estava errada

Comment (grvcoelho) [inline test/functional/boleto/register/pending.js]: :+1:

File: test/functional/boleto/register/pending.js [hunk 4]
```diff
@@ -38,4 +32,34 @@ test('registers a boleto (provider pending)', async (t) => {
...
+
+  restoreFunction(provider, 'register')
+})
+
+test('registers a boleto (provider timeout -> pending)', async (t) => {
+  mockFunction(
+    provider,
+    'register', () =>
+      Promise.resolve({ status: 'unknown' })
+      .delay(6000)
```
Comment (grvcoelho) [inline test/functional/boleto/register/pending.js]: Dá um <tab> aqui:

```
Promise.resolve({ status: 'unknown' })
  .delay(6000)
```

File: src/providers/bradesco/index.ts [hunk 5]
```diff
@@ -125,6 +126,10 @@ export const register = (boleto) => {
     .tap((response) => {
       logger.info({ status: 'succeeded', metadata: { status: response.status, data: response.data } })
     })
+    .catch(Promise.TimeoutError, () => {
+      logger.error({ status: 'failed', metadata: { error: 'Request timed out' } })
```
Comment (mccraveiro) [inline src/providers/bradesco/index.ts]: Linha muito longa. Quebra as propriedades em linhas

File: test/functional/boleto/register/pending.js [hunk 6]
```diff
@@ -1,23 +1,17 @@
...
-test.after(() => {
-  restoreFunction(provider, 'register')
-})
-
 const create = normalizeHandler(boletoHandler.create)
 
 const register = promisify(boletoHandler.register)
 
 test('registers a boleto (provider pending)', async (t) => {
+  mockFunction(provider, 'register', () => Promise.resolve({ status: 'unknown' }))
```
Comment (mccraveiro) [inline test/functional/boleto/register/pending.js]: Linha muito longa

**Pull request closed without merging**